### ISSUE TITLE
Feature Request: Astech/medics as green techs/docs vs separate entity

### ISSUE THREAD

**Issue by aschult1** [MegaMek/mekhq#542]
Is there a way to make all green techs/docs work like astechs or medics rather than having them separately? When they skill up to Regular, become "regular" working techs and docs?

**Comment by scJazz2** [comment]
You could adjust the Experience and Skills sections. This is in fact exactly what they are there to do.

The question is... why?

**Comment by Qwertronix** [comment]
Yeah this is pretty much already possible in the game currently, with a bit of manual skilling and job changes and such.

**Comment by PhoenyxCullen** [comment]
How would you suggest I do this? (btw, aschult1 is my school account).

**Comment by Qwertronix** [comment]
Generally, over time you will gain Astech "Dependent" personnel over the course of a standard Against-the-Bot campaign, and these Astech personnel can gain XP over time naturally or you can give them XP manually by turning on GM mode and right-clicking them (you can multi-select a bunch and right click to add XP to more than one at a time). 

Once they reach 5XP, they'll be able to buy a technical skill using the "Spend XP" option from the same right-click context menu, at which point they have the technical skill and can be switched to a primary "Technician" role if you want, or kept as Astechs until they earn another 10 XP at which point they can rank up the technical skill again to Regular, and then you switch their job. 

To do the reverse, find any "Green" rank technicians in your personnel list, use the Spend XP menu to buy the Astech skill (it's free), and then change their role to Astech. Keep them there until they earn the 10XP to rank up the tech skill, and once they have that just change them back.

Basically you'll have to manually look through the personnel list for any astechs or medics with >5 or >10 XP once in a while, but they earn XP pretty slowly.

**Comment by Kendricke** [comment]
I remember requested something like this a couple years ago, actually:

https://sourceforge.net/p/mekhq/feature-requests/292/

**Comment by purpledragon2** [comment]
The way I learned it back in the day was that astechs were supposed to be "techs in training" and medics were supposed to be "doctors in training"  I find merit in the request to be able to turn astechs into techs per xp.  Problem is that they don't get much xp.  Now, if there were a way to assign them to specific techs, maybe via a "tech squad" in the TO&E wherein they would not be counted as a combat unit...    Then, maybe they could get the same xp their tech they are assigned to gets...
I'm just trying to help see this through.

**Comment by PhoenyxCullen** [comment]
That would be better. Their lack of experience is their downfall. Actually make them teams so they can get more experience as a team when the so-called team lead gets a bonus xp during their tasks on a failure/success (make it a team effort).

**Comment by Qwertronix** [comment]
This would only work for Dependents and not temps in the pool... right? Because tracking this for temps would be a whole other thing as temps right now are just floating numbers with no proper person object for each one. 

I could definitely see having astechs and medics that are on permanently as dependents being "assigned" to a Tech or Doctor's team in the same way we assign other roles to units, though. And I believe that would be simpler and more intuitive in the current UI than doing this in the TO&E.

**Comment by PhoenyxCullen** [comment]
I think you hit it on the nose. Ever worked for a Temp Agency? :) Temporary Agents are workers from another corporation and are there as fill-ins where there none. Yes, temporarily assigned to a team, but they are extras that do not matter.

Thank you for bringing that help. How is this for a programming concept?

Lets take all the 'permanent'/dependent astechs and equally assign them to team leads so you get 12 leads with 3-4 astechs (depending on how many their are) that are on our financial books and fill-in with temporary across the board as needed from some temp agency.

**Comment by Qwertronix** [comment]
Hmm well since all dependent assistants enter the force at some point either through manual hiring or through the monthly "gain dependents" roll, I'd say that when they do so they join a random existing team that has free slots, if any, perhaps optionally with auto-balancing so that new ones always go to the team with the least permanent assistants? 

The player should still be able to override that and change an assistant to another team, IMO. This would use an "Assign to Team" right-click context menu option similar to Assign to Unit, but listing the names of techs/doctors whose teams have free positions for a dependent to fill. 

As @purpledragon2 said, they would earn xp as a team, and this could open up the interesting debate of whether we also want them to retire as a team when the team lead retires (similar to infantry). Perhaps, to prevent abuse on the last point, they would only earn XP from the team if they've been on that team at least one full cycle?

**Comment by PhoenyxCullen** [comment]
Maybe the team is assigned when you assign the Tech Lead to the Lance/Unit that they are doing maintenance for. Thus that team is fully responsible all the maintenance and repairs for that Mech. Real world example, a Military Engineer Company is responsible for a shop dealing with up to 10 vehicles a day, not all vehicles will get repaired in a day, but all of the vehicles will have had some work done on them during that day, as priority requires. Should we take that approach in a simplified Tech Team to TO&E Unit assignment?

**Comment by purpledragon2** [comment]
Just to be sure, they shouldn't have to be dependants.  You can hire astechs through the personnel market in AtB without using the temp options.  However, I do agree that the temps should not matter, unless there might be a later option to recruit temps over to permanent.  But that would be outside the scope of this discussion and, I think, the game.

**Comment by Qwertronix** [comment]
Oh, right, my bad. I simply implicitly associated any permanent assistant with being a dependent, but they don't have to be, yeah.

**Comment by pheonixstorm** [comment]
Another problem when using dependents is not every dependent will become a tech. In FMMr you can promote dependents to any skillset. Mechwarrior, ASF Pilot, vee gunner, soldier, etc. Not sure why AtB has all the dependents as astechs. Something to ask @neoancient about. I don't recall seeing anything like this in the AtB rules, though I could have missed it.

Astechs from the market are a different issue, but easier to deal with in the way you are asking. I say automating this for AtB dependents should be a no go. For hired astechs it would be fine.

**Comment by Necrofox91** [comment]
Well would it be possible to also "fold" the astechs/medics under the main tech/doctor in the personnel file? It would clean up the personnel tab quite nicely...
That gives me an idea about "folding" soldiers/battlearmor under the unit leader too, get that 250+ personnel tab down to size...
May think about opening an issue about that when it isn't this late and I can think clearly.

**Comment by jschmetzer** [comment]
+1 the idea of "tech teams" for those who actually want to hire named astechs/medics but can't bear the bloated Personnel view.

**Comment by IllianiCBT** [comment]
While an interesting idea, I don't see this request gaining any traction and it would require significant reworks to become viable. Closing.
